packages/cli/src/executions/__tests__/execution-recovery.service.test.ts

```diff
@@ -152,6 +152,31 @@ describe('ExecutionRecoveryService', () => {
 				expect(amendedExecution).toBeNull();
 			});
 
+			test('for errored dataful execution, should return `null`', async () => {
+				/**
+				 * Arrange
+				 */
+				const workflow = await createWorkflow();
+				const execution = await createExecution(
+					{ status: 'error', data: stringify({ runData: { foo: 'bar' } }) },
+					workflow,
+				);
+				const messages = setupMessages(execution.id, 'Some workflow');
```

Comment:
This test is not creating the 100% the same scenario you're covering in the PR description. It does contain an additional `n8n.workflow.started` message.